packages/docid/src/doc-id.ts

```diff
@@ -0,0 +1,157 @@
+import { CommitID } from './commit-id';
+import { DocRef } from './doc-ref';
+
+/**
+ * Parse DocID from bytes representation.
+ *
+ * @param bytes - bytes representation of DocID.
+ * @see [[DocID#bytes]]
+ */
+function fromBytes(bytes: Uint8Array): DocID {
+  const [docCodec, docCodecRemainder] = readVarint(bytes);
+  if (docCodec !== DOCID_CODEC) throw new Error('fromBytes: invalid docid, does not include docid codec');
+  const [docType, docTypeRemainder] = readVarint(docCodecRemainder);
+  const [cid] = readCid(docTypeRemainder);
+  return new DocID(docType, cid);
+}
+
+/**
+ * Parse DocID from string representation.
+ *
+ * @param input - string representation of DocID, be it base36-encoded string or URL.
+ * @see [[DocID#toString]]
+ * @see [[DocID#toUrl]]
+ */
+function fromString(input: string): DocID {
+  const protocolFree = input.replace('ceramic://', '').replace('/ceramic/', '');
+  const commitFree = protocolFree.includes('commit') ? protocolFree.split('?')[0] : protocolFree;
+  const bytes = multibase.decode(commitFree);
+  return fromBytes(bytes);
+}
+
+/**
+ * Document identifier, no commit information included.
+ * Encoded as '<multibase-prefix><multicodec-docid><doctype><genesis-cid-bytes>'
+ */
+export class DocID implements DocRef {
+  readonly #doctype: number;
+  readonly #cid: CID;
+
+  static fromBytes = fromBytes;
+  static fromString = fromString;
+
+  /**
+   * Create a new DocID.
+   *
+   * @param {string|number}      doctype
+   * @param {CID|string}         cid
+   *
+   * @example
+   * ```typescript
+   * new DocID('tile', 'bagcqcerakszw2vsovxznyp5gfnpdj4cqm2xiv76yd24wkjewhhykovorwo6a');
+   * new DocID('tile', cid);
+   * new DocID(0, cid);
+   * ```
+   */
+  constructor(doctype: string | number, cid: CID | string) {
+    if (!(doctype || doctype === 0)) throw new Error('constructor: doctype required');
+    if (!cid) throw new Error('constructor: cid required');
+    this.#doctype = typeof doctype === 'string' ? doctypes.indexByName(doctype) : doctype;
+    this.#cid = typeof cid === 'string' ? new CID(cid) : cid;
+  }
+
+  /**
+   * Doc type code
+   */
+  get type(): number {
+    return this.#doctype;
+  }
+
+  /**
+   * Doc type name
+   */
+  get typeName(): string {
+    return doctypes.nameByIndex(this.#doctype);
+  }
+
+  /**
+   * Genesis record CID
+   */
+  get cid(): CID {
+    return this.#cid;
+  }
+
+  /**
+   * Bytes representation of DocID.
+   */
+  @Memoize()
+  get bytes(): Uint8Array {
+    const codec = varint.encode(DOCID_CODEC);
+    const doctype = varint.encode(this.type);
+
+    return uint8ArrayConcat([codec, doctype, this.cid.bytes]);
+  }
+
+  /**
+   * Copy of self. Exists to maintain compatibility with CommitID.
+   * @readonly
+   */
+  get baseID(): DocID {
+    return new DocID(this.#doctype, this.#cid);
+  }
+
+  /**
+   * Construct new CommitID for the same document, but a new `commit` CID.
+   */
+  travel(commit: CID | string | number): CommitID {
```

Comment:
`toCommitID`?

Reply:
For DocID it is perfect. For CommitID, I feel like `commitId.toCommitID` is weird, and seems like something surprising.

Something like this goes across my mind: "Here is commit id, we transform it into another commit id. Commit id - to commit id. Hmm, it must be something fishy then"

If it doesn't cause a similar effect on you, I will change the name.

Reply:
Do they need to have the same name in both classes? Could be `toOtherCommitID` in the CommitID class?

Reply:
atCommit maybe??

Reply:
Makes sense to me 👍